src/main/java/com/dataguise/saas/service/OverviewService.java

```diff
@@ -0,0 +1,316 @@
+        List<OverviewSourceDetail> databaseTreeList = new ArrayList<>();
+        try {
+            databaseList = restDriver.getAllDatabasesInfoForHost(hostNamePortNumberStruct);
+            for (String database : databaseList) {
+                OverviewSourceDetail overviewSourceDetail = new OverviewSourceDetail();
+                overviewSourceDetail.setLabel(database);
+                overviewSourceDetail.setSourceType(Modules.RDS.name());
+                databaseTreeList.add(overviewSourceDetail);
+            }
+            return databaseTreeList;
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+
+    }
+
+    /**
+     * Get dbms cloud tables for a database of a selected hostname from controller api.
+     *
+     * @return a  list of tables within a database .
+     */
+
+    public List<OverviewSourceDetail> getAllTablesInfoForHost(HostNamePortNumberStruct hostNamePortNumberStruct, String database) throws BadGatewayException, InternalServerException {
+
+        List<String> tableList = new ArrayList<>();
+        List<OverviewSourceDetail> tableTreeList = new ArrayList<>();
+        try {
+            tableList = restDriver.getAllTablesInfoForHost(hostNamePortNumberStruct, database);
+            for (String table : tableList) {
+                OverviewSourceDetail overviewSourceDetail = new OverviewSourceDetail();
+                overviewSourceDetail.setLabel(table);
+                overviewSourceDetail.setSourceType(Modules.RDS.name());
+                tableTreeList.add(overviewSourceDetail);
+            }
+            return tableTreeList;
+
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+
+    }
+
+    /**
+     * Get dbms cloud Columns Info for a database of a selected hostname from controller api.
+     *
+     * @return column details within a table .
+     */
+
+    public DgHostColumnsInfoWrapper getAllColumnsInfoForHost(HostNamePortNumberStruct hostNamePortNumberStruct, String database, String tableName) throws BadGatewayException, InternalServerException {
+
+        DgHostColumnsInfoWrapper dgHostColumnsInfoWrapper = new DgHostColumnsInfoWrapper();
+        try {
+            dgHostColumnsInfoWrapper = restDriver.getAllColumnsInfoForHost(hostNamePortNumberStruct, database, tableName);
+            return dgHostColumnsInfoWrapper;
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+
+    }
+
+    /**
+     * Get lfaS3 cloud cluster result Info for a directories and files from controller api.
+     *
+     * @return column details within a table .
+     */
+
+    public DgS3OverviewResultsWrapper getS3OverviewResultsDirs(Integer clusterId, String dirFullPath) throws BadGatewayException, InternalServerException {
```

Comment:
**Reporter**: Checkstyle    
    **Rule**: com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMethodCheck
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/OverviewService.java L299
    
    Expected @param tag for &#39;clusterId&#39;.
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-841194080>*